apps-api/backoffice/src/modules/admin/announcements/index.ts

```diff
@@ -0,0 +1,379 @@
+        }
+      }
+
+      return status(200, result.value)
+    },
+    {
+      requiredUser: true,
+      params: AnnouncementIdParams,
+      body: PutDraftedAnnouncementBody,
+      response: {
+        200: PutDraftedAnnouncementResponse,
+        ...createErrorSchema(
+          InternalErrorCode.ANNOUNCEMENT_NOT_FOUND,
+          InternalErrorCode.INTERNAL_SERVER_ERROR
+        ),
+      },
+      detail: {
+        summary: 'Update drafted announcement by ID',
+        description: 'Update a specific drafted announcement by its ID',
+      },
+    }
+  )
+  .post(
+    '/:announcementId/publish',
+    async ({ params, user, status, adminAnnouncementService }) => {
+      const result = await adminAnnouncementService.publishDraftedAnnouncementById(
+        params.announcementId,
+        user.sub
+      )
+      if (result.isErr()) {
+        switch (result.error.code) {
+          case InternalErrorCode.ANNOUNCEMENT_NOT_FOUND:
+            return mapErrorCodeToResponse(result.error, status)
+          case InternalErrorCode.INTERNAL_SERVER_ERROR:
+            return mapErrorCodeToResponse(result.error, status)
+          default:
+            exhaustiveGuard(result.error)
+        }
+      }
+
+      return status(200, result.value)
+    },
+    {
+      requiredUser: true,
+      params: AnnouncementIdParams,
+      response: {
+        200: DraftedAnnouncementPublishedResponse,
+        ...createErrorSchema(
+          InternalErrorCode.ANNOUNCEMENT_NOT_FOUND,
+          InternalErrorCode.INTERNAL_SERVER_ERROR
+        ),
+      },
+      detail: {
+        summary: 'Publish drafted announcement by ID',
+        description: 'Publish a specific drafted announcement by its ID',
+      },
+    }
+  )
+  .delete(
+    '/:announcementId',
+    async ({ params, status, adminAnnouncementService }) => {
+      const result = await adminAnnouncementService.deleteDraftedAnnouncement(params.announcementId)
+      if (result.isErr()) {
+        switch (result.error.code) {
+          case InternalErrorCode.ANNOUNCEMENT_NOT_FOUND:
+            return mapErrorCodeToResponse(result.error, status)
+          case InternalErrorCode.INTERNAL_SERVER_ERROR:
+            return mapErrorCodeToResponse(result.error, status)
+          default:
+            exhaustiveGuard(result.error)
+        }
+      }
+
+      return status(200, result.value)
+    },
+    {
+      requiredUser: true,
+      params: AnnouncementIdParams,
+      response: {
+        200: DeleteDraftedAnnouncementResponse,
+        ...createErrorSchema(
+          InternalErrorCode.ANNOUNCEMENT_NOT_FOUND,
+          InternalErrorCode.INTERNAL_SERVER_ERROR
+        ),
+      },
+      detail: {
+        summary: 'Delete drafted announcement by ID',
+        description: 'Remove a specific drafted announcement by its ID',
+      },
+    }
+  )
+
+export const AdminAnnouncementsController = new Elysia({
+  prefix: '/announcements',
+  adapter: node(),
```

Comment:
Just an optional one, could you remove node adapter from controller, service, repository ?